python-checks/src/main/java/org/sonar/python/checks/LocalVariableAndParameterNameConventionCheck.java

```diff
@@ -78,7 +83,42 @@ private void checkName(Symbol symbol, SubscriptionContext ctx) {
   }
 
   private static boolean isType(Symbol symbol) {
-    return symbol.usages().stream().map(Usage::tree).filter(Expression.class::isInstance).map(Expression.class::cast).anyMatch(e -> e.type().mustBeOrExtend("type"));
+    return isExtendingType(symbol) || isAssignedFromSpecialForm(symbol);
+  }
+
+  private static boolean isExtendingType(Symbol symbol) {
+    return symbol.usages().stream().map(Usage::tree).filter(Expression.class::isInstance).map(Expression.class::cast).anyMatch(e -> e.type().mustBeOrExtend("type")) ||
+      Objects.equals(symbol.annotatedTypeName(), "typing._SpecialForm");
```

Comment:
_Nitpick:_ to avoid the `null` case, you can also write `"typing._SpecialForm".equals(symbol.annotatedTypeName())` which is a bit more concise.

Another more conservative approach could've been to simply exclude any symbol whose annotated type name starts with `typing`, as the likelihood of a type alias is high in such case, and it would avoid to single out `_SpecialForm` explicitly.

Reply:
I follow your idea to exclude any symbol whose annotated type starts with `typing` :smile: